beets/autotag/match.py

```diff
@@ -208,6 +208,10 @@ def track_distance(
     if item.mb_trackid:
         dist.add_expr("track_id", item.mb_trackid != track_info.track_id)
 
+    # Penalize mismatching disc numbers.
+    if track_info.medium and item.disc:
+        dist.add_expr("medium_index", item.disc != track_info.medium)
```

Comment:
```suggestion
        dist.add_expr("medium", item.disc != track_info.medium)
```

I think you'd want to call this **medium** here, as `medium_index` is being checked above (confusingly) as `track_index`:

```py

def track_index_changed(item: Item, track_info: TrackInfo) -> bool:
    """Returns True if the item and track info index is different. Tolerates
    per disc and per release numbering.
    """
    return item.track not in (track_info.medium_index, track_info.index)

    ...

    if track_info.index and item.track:
        dist.add_expr("track_index", track_index_changed(item, track_info))

```

Reply:
Thanks for taking a look! I think it is a bit confusing that it's called `disc` on the item and `medium` on the track_info. I think I was trying to create something new, but missed that `medium_index` is already an attribute of `TrackInfo`. I was worried that `medium` might imply that the mismatch was on the media type and not the media "number" but it is already used that way in `TrackInfo` so I think this is fine too.

Reply:
Yeah I agree all this naming is far too confusing. To this day I still need to look at the data to remember what each term means:
| **`media`** | **`disctotal`** / **`mediums`** | **`disc`** / **`medium`** | **`tracktotal`** / **`medium_total`** | **`track_alt`** | **`track`** / **`medium_index`** |          **`title`**           |
| :---------: | :-----------------------------: | :-----------------------: | :-----------------------------------: | :-------------: | :------------------------------: | :----------------------------: |
|    Vinyl    |                2                |             1             |                   3                   |       A1        |                1                 | Oh, Lovely Appearance of Death |
|    Vinyl    |                2                |             1             |                   3                   |       A2        |                2                 |           Good Winds           |
|    Vinyl    |                2                |             1             |                   3                   |        B        |                3                 |          New Atlantis          |
|             |                                 |                           |                                       |                 |                                  |                                |
|    Vinyl    |                2                |             2             |                   5                   |       C1        |                1                 |     At The Strangers House     |
|    Vinyl    |                2                |             2             |                   5                   |       C2        |                2                 |         A Land Unknown         |
|    Vinyl    |                2                |             2             |                   5                   |       C3        |                3                 |             Temple             |
|    Vinyl    |                2                |             2             |                   5                   |       D1        |                4                 |           Black Sun            |
|    Vinyl    |                2                |             2             |                   5                   |       D2        |                5                 |        The Sound House         |